addons/account/models/account_move.py

```diff
@@ -4206,6 +4215,62 @@ def _compute_tax_line_id(self):
 
+    @api.depends('tax_line_id.tax_exigibility', 'tax_ids.tax_exigibility')
+    def _compute_all_taxes_are_on_payment(self):
+        for record in self:
+            record.has_taxes = (record.tax_line_id or record.tax_ids)
+            record.all_taxes_are_on_payment = (
+                (not record.tax_line_id or record.tax_line_id.tax_exigibility == "on_payment")
+                and all(tax.tax_exigibility == "on_payment" for tax in record.tax_ids)
+            )
+
+    @api.depends('tax_ids', 'move_id.line_ids')
+    def _compute_tax_move_line_ids(self):
+        # prefetch
+        self.mapped('move_id.line_ids')
+        def endswith(a, b):
+            for i, x in enumerate(reversed(a)):
+                if b[-1-i] != x:
+                    return False
+            return True
+
+        def line2taxes(line):
+            return (
+                (line.tax_ids | line.tax_ids.children_tax_ids)
+                .filtered(lambda tax: tax.amount_type != "group" and tax.is_base_affected)
+                .sorted(lambda tax: (tax.sequence, tax.id))
+            )
+
+        for base_line in self:
+            tax_line_ids = []
+            if not base_line.tax_repartition_line_id:
+                tax_line_ids = base_line.move_id.line_ids.filtered(
+                    lambda tax_line:
+                        tax_line.tax_line_id
+                        and tax_line.tax_repartition_line_id
+                        and (tax_line.group_tax_id or tax_line.tax_line_id) in base_line.tax_ids
+                        # see https://github.com/odoo/odoo/commit/48d60151254045768d5aac1b29c5acf84b70cef2
+                        and (tax_line.move_id.move_type != 'entry' or tax_line.balance * base_line.balance * tax_line.tax_line_id.amount * tax_line.tax_repartition_line_id.factor_percent > 0)
+                        and base_line.partner_id == tax_line.partner_id  # TODO: is it needed?
+                        and base_line.currency_id == tax_line.currency_id  # TODO: is it needed?
+                        and ((tax_line.tax_repartition_line_id.account_id or base_line.account_id) == tax_line.account_id
+                             or tax_line.tax_line_id.tax_exigibility == 'on_payment' and tax_line.tax_line_id.cash_basis_transition_account_id
+                        )
+                        and (
+                            not tax_line.tax_line_id.analytic
+                            or not base_line.analytic_account_id and not tax_line.analytic_account_id
+                            or base_line.analytic_account_id == tax_line.analytic_account_id
+                        )
+                        and (
+                            not tax_line.tax_line_id.include_base_amount
+                            or endswith(
+                                tax_line.tax_line_id | line2taxes(tax_line),
+                                line2taxes(base_line)
+                            )
```

Comment:
I have came across your PR.
I didn't read everything as I get the essence but I have the intuition that this is equivalent to the join here: https://github.com/odoo/odoo/blob/d40a118250503cd03d0ac4c58839521d82a466db/addons/account/models/account_move_line_tax_details.py#L168-L198

If it's the case, take into consideration that I made a bugfix https://github.com/odoo/odoo/commit/48d60151254045768d5aac1b29c5acf84b70cef2 adding one more condition to handle misc entry.

Reply:
> I'm sure to understand everything

That's a bold statement :smile: